numba/cpython/builtins.py

```diff
@@ -298,6 +298,14 @@ def int_impl(context, builder, sig, args):
     return impl_ret_untracked(context, builder, sig.return_type, res)
 
 
+@lower_builtin(float, types.StringLiteral)
+def float_inf_impl(context, builder, sig, args):
+    [ty] = sig.args
+    assert ty.literal_value.lower() in ('inf', '-inf')
+    res = context.get_constant(sig.return_type, np.inf)
```

Comment:
I guess this should depend on the sign of the input?

Reply:
Yes! My bad here. Will fix this in a bit.